src/plugin/settings/engine.cpp

```diff
@@ -204,242 +206,243 @@ Result<const ISettingsResponse *> Engine::settingsReceived()
-        const std::string mountDir = cfManager.getMountDir();
-        const std::string fileCacheDir = cfManager.getFileCacheDir();
-        std::string passphrase = "";
-        // Generate passphrase for config file
-        unsigned char key[32]; // AES-256 key
-        if (RAND_bytes(key, sizeof(key)))
-        {
-            // Need to encode passphrase to base64 to pass to cloudfuse
-            BIO *bmem, *b64;
-            BUF_MEM *bptr;
-
-            b64 = BIO_new(BIO_f_base64());
-            bmem = BIO_new(BIO_s_mem());
-            b64 = BIO_push(b64, bmem);
-            BIO_set_flags(b64, BIO_FLAGS_BASE64_NO_NL);
-            BIO_write(b64, key, 32);
-            BIO_flush(b64);
-            BIO_get_mem_ptr(b64, &bptr);
-
-            passphrase = std::string(bptr->data, bptr->length);
-        }
-        else
-        {
-            NX_PRINT << "OpenSSL Error: Unable to generate secure passphrase";
-            return error(ErrorCode::internalError, "OpenSSL Error: Unable to generate secure passphrase");
-        }
+        NX_PRINT << "Settings changed.";
+        mount();
+    }
+    else
+    {
+        NX_PRINT << "Settings have not changed.";
+    }
 
-        std::error_code errCode;
+    if (!processActiveSettings(&model, &values))
+    {
+        std::string errorMessage = "Unable to process the active settings section";
+        NX_PRINT << errorMessage;
+        return error(ErrorCode::internalError, errorMessage);
+    }
 
-        // Create mount directory if it does not exist
-        NX_PRINT << "Creating mount directory if it does not exist";
-        if (fs::exists(mountDir))
-        {
-            // Unmount before mounting
-            const processReturn unmountReturn = cfManager.unmount();
-            if (unmountReturn.errCode != 0)
-            {
-                return error(ErrorCode::internalError, "Failed to unmount. Here's why: " + unmountReturn.output);
-            }
+    auto settingsResponse = new SettingsResponse();
+    settingsResponse->setModel(makePtr<String>(Json(model).dump()));
+    settingsResponse->setValues(makePtr<StringMap>(values));
 
-#if defined(__linux__)
-            // On Linux we need to check the folder has the correct permissions
-            fs::file_status s = fs::status(mountDir);
-            if ((s.permissions() & fs::perms::all) != fs::perms::all)
-            {
-                fs::permissions(mountDir, fs::perms::all, fs::perm_options::add, errCode);
-                if (errCode)
-                {
-                    NX_PRINT << "Unable to set mount directory permission with error: " + errCode.message();
-                    return error(ErrorCode::internalError,
-                                 "Unable to set mount directory permission with error: " + errCode.message());
-                }
-            }
-#endif
-        }
-        else
+    return settingsResponse;
+}
+
+bool Engine::mount()
+{
+    // gather mount options
+    getMountOptions();
```

Comment:
I'm not in love with the getMountOptions function. Since it has side effects that need to set the member variables so that they can be updated and used later in the mount. I also kind of disagree that they should be member variables instead of method parameters. Since, when we mount we also get the latest version of those options from the VMS, we never need to read these values again so I don't think it makes sense to have them be a part of the class. I would rather set those here, and then pass them as parameters to validateMount and spawnMount. But I am open to discussion on the thought process here.

Reply:
Originally, I had `validateMount` and `spawnMount` both extract those values from the settings map.
And then I just had a feeling that I didn't like that block of code repeated, so I tore it out and replaced it with members.
I see what you're saying about unnecessary members and side effects. I'll put it back the way it was before, and maybe we can discuss from there.

Reply:
Yeah that makes sense. I mean it could be moved into mount to prevent that type of duplication, and then passed to validate and spawn, but it isn't too much duplicated code since it is just reading the values.

Reply:
Yeah, especially now that I see that `spawnMount` only needs a few of the values, I like this way better. I'll leave it as it (now) is.